### ISSUE TITLE
R-Extension fails to find user.properties

### ISSUE THREAD

**Issue by cstaelin** [NetLogo/R-Extension#15]
The latest version of the R-Extension for NL 6.1.0 on Windows 10 looks for the user.properties file in extensions\r rather than in extensions\.bundled\r.  The problem in evident in Configuration.java, line 69
```
Path userPropertiesPath =
        ExtensionManager$.MODULE$.extensionsPath().resolve("r").resolve("user.properties");
```
where it should be
```
Path userPropertiesPath = 
ExtensionManager$.MODULE$.extensionsPath().resolve(".bundled").resolve("r").resolve("user.properties");
```

The workaround is simply to put user.properties where its expected.

**Comment by LaCuneta** [comment]
Thanks for reporting this issue.

This was fixed in [this commit](https://github.com/NetLogo/R-Extension/commit/858023fe636d6ac92523ae7295b7c8a944838406).  The resulting fix is published to the library for [the extensions manager](https://ccl.northwestern.edu/netlogo/docs/extension-manager.html) as version 1.2.2.  Once installed, you'll need to place the `user.properties` file in a new, user-specific location, per [the new documentation](https://github.com/NetLogo/R-Extension#configuring-the-r-extension).  

Let me know if you have any difficulties with it.

**Comment by cstaelin** [comment]
Jeremy,

This has not solved the problem for me.  First, when I renamed the
.bundled\r directory to force NetLogo to download the extension, it
downloaded the old version (at least from the dates of he files,
05/17/2019).  So I went to NetLogo/Libraries and downloaded r-1.2.2.zip and
put that (unzipped) in the .bundled\r directory.  The dates on those files
are all 07/23.  However, I still get the old error, no matter where I put
user.properties.  "Could not find R_Home.  Please set R home in the
environment or in user.properties (Error #01)"  It does not tell me where
to put user.properties as the new documentation suggests it should.  I am
closing NetLogo between all these tests, so it should be loading the now
newer version.  Am I doing something wrong?  (I tried putting
user.properties in the same directory as the model I'm opening, but that
didn't seem to help.)

Thanks,
Charles

-------------------------------
Charles P. Staelin
Professor of Economics
Smith College
Northampton, MA 01063

**Comment by LaCuneta** [comment]
Sorry it's still giving you problems!

To start with, you shouldn't have to do too much fiddling with the `.bundled` extensions folder.  The extension manager will not update extensions that live there, it will place updates them [in a user profile-specific folder](https://ccl.northwestern.edu/netlogo/docs/extensions.html#where-extensions-are-located) - you won't see any changes in the NetLogo directory.  The extension manager should also handle updates for you without any manual work like renaming/delete folders, if [you open it up](https://ccl.northwestern.edu/netlogo/docs/extension-manager.html) it should show you the extensions with updates available to install.

I do see the new `user.properties` path isn't correctly showing in some error messages for the R extension, I'll get that updated and a new release put out.

Can I ask - are you on Windows?  The correct path for `user-properties` would then be something like `C:\Users\USER_NAME\AppData\Roaming\NetLogo\6.1\r`, depending on your Windows version.  For macOS it'd be `/users/USER_NAME/Library/Application Support/NetLogo/6.1/r`

**Comment by cstaelin** [comment]
Jeremy,

First, I apologize for not having really taken careful note of the new
extension manager documentation.  I see now how it is supposed to work, and
I tested it with my own stats extension first.  Worked beautifully, except
that the documentation is slightly off - in Windows the extensions are
placed in AppData\*Roaming*\NetLogo\6.1\extensions - the documentation
leaves out the "Roaming" .

Then I eliminated all traces of the r extension from my hard drive and
started up my test model.  NetLogo immediately went into its install mode,
downloaded the r extension, and gave the the same old error.  I checked,
and saw that it had placed version 1.2.1 in the \.bundled directory,
confirmed by looking at the file dates and at the extension manager (under
tools), which said that 1.2.1 was installed, but that 1.2.3 was available.
It had also created an empty directory at AppData\Roaming\NetLogo\6.1\r,
but nothing in the extensions folder in that tree.  So, I then asked the
extension manager to update the r extension, which is did nicely, putting
the new version where it belonged in
AppData\Roaming\NetLogo\6.1\extensions\r, and the test model gave me the
correct error message to put user.properties in
AppData\Roaming\NetLogo\6.1\r.  After doing that, all worked perfectly.

So, the only problem I see is that the initial install was of the wrong
version. I have no idea if that was just me, but perhaps it's worth
checking.

Thanks very much for the updates and the guidance.

Best,
Charles

-------------------------------
Charles P. Staelin
Professor of Economics
Smith College
Northampton, MA 01063
